### ISSUE TITLE
Function keys not working on OSX

### ISSUE THREAD

**Issue by clyra** [houmain/keymapper#172]
Hi all,

It seems that functions keys are not working or being misinterpreted by keymapper. At first I noticed that the volume keys are not working. Investigating further and using keymapperctl --next-key-info, I'm getting this:

volume up (without pressing fn): nothing

volume up (pressing fn)
key_name = 'Space'
scan_code = 0x002C
system = 'MacOS'
class = 'kitty'
path = '/Applications/kitty.app/Contents/MacOS/kitty'
device = 'Keyboard K380        

volume down (without pressing fn): nothing
volume down (pressing fn):
key_name = 'F12'
scan_code = 0x0045
system = 'MacOS'
class = 'kitty'
path = '/Applications/kitty.app/Contents/MacOS/kitty'
device = 'Keyboard K380

F5 (without pressing fn):
key_name = 'ArrowUp'
scan_code = 0x0052

F5 (pressing fn):
key_name = 'F5'
scan_code = 0x003E

So, some keys looks ok, but most not. I'm running keymapperctl 4.6.0, slight modified to ignore the mx anywhere 2s mouse.

**Comment by ristomatti** [comment]
Please check if comments on #120 have anything useful.

**Comment by houmain** [comment]
Thanks for reporting. I will try to reproduce it soon.

**Comment by clyra** [comment]
Thanks for link to link to the previous issue involving functions keys!

I tried to remapping the function keys but had partial success as I had to press the fn key to access the "multimedia" keys, and the mapping was wrong, ie, mute mapped to vol down, vol down mapped to vol up, and vol up still does nothing. I guess that the keyboard (logitech K380) itself does some "mapping" because even using the event viewer from karabiner i cannot see the "fn" event.  F12 maps to "consumer_key_code":  "volume decrement" or to F12 if fn key is pressed too.  The next key is kind weird, because it shows "consumer_key_code": "volume increment" (keycode 0x00E9) or Lcontrol + Spacebar if Fn key is pressed.

It seems that 0x00E9 is not mapped in key.h. Please, let me know how I can help/test this mapping.

**Comment by clyra** [comment]
> I could finally investigate the issue. This problem occurs when Karabiner-Elements grabs the physical device and then Keymapper grabs the virtual device created by Karabiner-Elements. I do not have a solution yet (but to stop Karabiner-Elements from grabbing the device).

Thanks! I guess I didnt fully understood, but by simply closing karabiner elements app shouldnt avoid this? Or there's also a daemon/service that should be stopped too? Or worse, and it cant be avoided?

**Comment by houmain** [comment]
>  but by simply closing karabiner elements app shouldnt avoid this?

Yes, Quiting it or deactivating the keyboard in the Devices section should hopefully fix the problem (a reboot may be required).

**Comment by houmain** [comment]
This should be fixed now with 44c379fdff2efdac65dea85f9fa1f3b5ebd5bcc7.
